docs/zdgbook/ComponentsAndInterfaces.rst

```diff
@@ -268,39 +262,44 @@ method.
 
 For example::
 
-  >>> IUser.namesAndDescriptions()
-  [('getUserName', <zope.interface.interface.Method.Method object at 80f38f0>),
-  ('getFavoriteColor', <zope.interface.interface.Method.Method object at 80b24f0>),
-  ('getPassword', <zope.interface.interface.Method.Method object at 80fded8>)]
+  >>> IHello.namesAndDescriptions()
+dict_items([('hello', <zope.interface.interface.Method object at 0x7fc6875110f0>)])
 
-As you can see, the "description" of the Interface's three items in
-these cases are all `Method` objects.  Description objects can be
-either 'Attribute' or `Method` objects.  Attributes, methods, and
-interface objects implement the following interface::
+.. note::
+  You cannot access the `Method` object by index, as
+  ``namesAndDescriptions`` returns a dict_view.
```

Comment:
But only on Python 3. I wonder if there's an argument to be made that returning lists is actually the most useful behaviour on all versions of Python. Returning views may save memory, but these are small objects; returning views may stay in sync, but these objects don't change.